### ISSUE TITLE
Cargo panics when attempting to compile project on shared partition

### ISSUE THREAD

**Issue by tdoylend** [rust-lang/rust#52440]
I have a machine which dual-boots between Windows and Linux (Mint 18.3.) Rust is installed on both platforms. A shared partition (formatted Ext4 and accessed on Windows via Ext2Fsd) holds current projects, for easy inter-platform access.

I created a Rust project on the shared partition in Linux (just the default "Hello World" example), and compiled and ran it without errors. On rebooting to Windows and attempting to compile again, however, the following error was reported:

```$ cargo run
warning: could not canonicalize path: 'J:\projects\realmworld\realmclient'
warning: could not canonicalize path: 'J:\projects\realmworld\realmclient'
warning: could not canonicalize path: 'J:\projects\realmworld'
warning: could not canonicalize path: 'J:\projects'
warning: could not canonicalize path: 'J:\'
   Compiling realmclient v0.1.0 (file:///J:/projects/realmworld/realmclient)
error: incremental compilation: error canonicalizing path 
`J:\projects\realmworld\realmclient\target\debug\incremental\realmclient-2s66szn1ud36l`: The system cannot find the file specified. (os error 2)

thread 'main' panicked at 'librustc\session\mod.rs:795: Trying to get session directory from IncrCompSession `NotInitialized`', librustc\session\mod.rs:1281:26
note: Run with `RUST_BACKTRACE=1` for a backtrace.
error: aborting due to previous error


error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.27.1 (5f2b325f6 2018-07-07) running on x86_64-pc-windows-gnu

note: compiler flags: -C debuginfo=2 -C incremental --crate-type bin

note: some of the compiler flags provided by cargo are hidden

error: Could not compile `realmclient`.

To learn more, run the command again with --verbose.

```

**Comment by Enselic** [comment]
Triage: This seems a like a duplicate to #48249 in practice, so closing as such.
